llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -795,6 +796,10 @@ processInputModule(std::unique_ptr<Module> M) {
   if (isModuleUsingAsan(*M))
     Modified |= runModulePass<SanitizeDeviceGlobalPass>(*M);
 
+  // Transform Joint Matrix builtin calls to align them with SPIR-V friendly
```

Comment:
Is there a reason why we need to invoke this pass inside sycl-post-link? Can we not run this pass in the standard LLVM pipeline?

Thanks

Reply:
I'm open to suggestions if something matches better, but the considerations are the following:
1. If optimizations are enabled I want this transformation to happen in the end of the pipeline;
2. If optimizations are disabled I want this transformation anyway.

In both cases sycl-post-link seems like a good match.

Reply:
I think this is reasonable. We will eventually be trying to refactor sycl-post-link anyways.
Thanks Dmitry

Reply:
> I'm open to suggestions if something matches better, but the considerations are the following:
> 
> 1. If optimizations are enabled I want this transformation to happen in the end of the pipeline;
> 2. If optimizations are disabled I want this transformation anyway.
> 
> In both cases sycl-post-link seems like a good match.

Both can be achieved with a standard optimization pipeline, we have a whole group of such passes: https://github.com/intel/llvm/blob/7f9e251d1934787ff9a71a0334e12288c92bf02e/clang/lib/CodeGen/BackendUtil.cpp#L1125-L1166